processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -1102,6 +1138,7 @@ public ColumnCapabilitiesImpl getCapabilities()
      * {@link RollupFactsHolder} needs concurrent collections, that are not present in fastutil.
      */
     private int rowIndex;
+    private long dimsKeySize;
```

Comment:
`dimsKeySize` could be final. And anything that could be final, should be final.

Reply:
Hmm, looks like i cannot make `dimsKeySize` final unless I add it as argument in all constructors, but I am leaning towards using private-constructor-plus-static-creator-methods pattern.